circleci: failed tests (14169): TestClientNotReady

The following test appears to have failed:

[#14169](https://circleci.com/gh/cockroachdb/cockroach/14169):

```
I160309 21:28:24.196167 /go/src/google.golang.org/grpc/clientconn.go:318  Failed to dial 127.0.0.1:43486: grpc: the client connection is closing; please retry.
--- PASS: TestUnretryableError (0.02s)
=== RUN   TestClientNotReady
W160309 21:28:24.219802 kv/send.go:195  error reply: kv/send.go:259: rpc to 127.0.0.1:37458 failed: context deadline exceeded
W160309 21:28:24.241996 kv/send.go:195  error reply: rpc error: code = 13 desc = "transport is closing"
--- FAIL: TestClientNotReady (0.08s)
	send_test.go:299: failed to send RPC: too many errors encountered (1 of 1 total): rpc error: code = 13 desc = "transport is closing"
=== RUN   TestComplexScenarios
W160309 21:28:24.304280 kv/send.go:195  error reply: failed to send RPC: test
W160309 21:28:24.304711 kv/send.go:195  error reply: failed to send RPC: test
W160309 21:28:24.304757 kv/send.go:195  error reply: failed to send RPC: test
W160309 21:28:24.304806 kv/send.go:195  error reply: failed to send RPC: test
W160309 21:28:24.304878 kv/send.go:195  error reply: failed to send RPC: test
W160309 21:28:24.305292 kv/send.go:195  error reply: failed to send RPC: test
W160309 21:28:24.305334 kv/send.go:195  error reply: failed to send RPC: test
W160309 21:28:24.305730 kv/send.go:195  error reply: failed to send RPC: test
--
W160309 21:28:44.972268 kv/dist_sender.go:627  failed to invoke BeginTransaction ["a",/Min), Put ["a",/Min), Put ["b",/Min): txn "kv/dist_sender_server_test.go:580 TestPropagateTxnOnPushError" id=576010de key="a" rw=true pri=0.03126473 iso=SERIALIZABLE stat=PENDING epo=1 ts=1457558924.970519720,0 orig=1457558924.970519720,0 max=1457558924.970519720,0 failed to push "kv/dist_sender_server_test.go:554 TestPropagateTxnOnPushError.func1" id=3898f7c9 key="b" rw=false pri=0.03126478 iso=SERIALIZABLE stat=PENDING epo=0 ts=1457558924.967168619,0 orig=1457558924.967168619,0 max=1457558924.967168619,0
I160309 21:28:44.972552 client/db.go:422  failed batch: txn "kv/dist_sender_server_test.go:580 TestPropagateTxnOnPushError" id=576010de key="a" rw=true pri=0.03126473 iso=SERIALIZABLE stat=PENDING epo=2 ts=1457558924.970519720,0 orig=1457558924.970519720,0 max=1457558925.020519720,0 failed to push "kv/dist_sender_server_test.go:554 TestPropagateTxnOnPushError.func1" id=3898f7c9 key="b" rw=false pri=0.03126478 iso=SERIALIZABLE stat=PENDING epo=0 ts=1457558924.967168619,0 orig=1457558924.967168619,0 max=1457558924.967168619,0
I160309 21:28:45.021999 storage/replica_command.go:453  auto-gc'ed 3898f7c9 (1 intents)
I160309 21:28:45.028022 storage/engine/rocksdb.go:141  closing in-memory rocksdb instance
--- PASS: TestPropagateTxnOnPushError (0.17s)
FAIL
FAIL	github.com/cockroachdb/cockroach/kv	23.112s
=== RUN   TestCombinable
--- PASS: TestCombinable (0.00s)
=== RUN   TestBatchSplit
--- PASS: TestBatchSplit (0.00s)
=== RUN   TestBatchRequestGetArg
--- PASS: TestBatchRequestGetArg (0.00s)
=== RUN   TestKeyNext
--- PASS: TestKeyNext (0.00s)
=== RUN   TestKeyPrefixEnd
--- PASS: TestKeyPrefixEnd (0.00s)

```
Please assign, take a look and update the issue accordingly.

--- a/kv/send_test.go
+++ b/kv/send_test.go
@@ -19,7 +19,6 @@ package kv
 import (
 	"errors"
 	"net"
-	"strings"
 	"testing"
 	"time"
 
@@ -272,18 +271,16 @@ func TestClientNotReady(t *testing.T) {
 	opts.Timeout = 0
 	c := make(chan error)
 	go func() {
-		if _, err := sendBatch(opts, []net.Addr{ln.Addr()}, nodeContext); err == nil {
-			c <- util.Errorf("expected error when client is closed")
-		} else if !strings.Contains(err.Error(), "failed as client connection was closed") {
-			c <- err
+		if _, err := sendBatch(opts, []net.Addr{ln.Addr()}, nodeContext); !testutils.IsError(err, "(failed as client connection was closed|transport is closing)") {
+			c <- util.Errorf("unexpected error when client was closed: %v", err)
 		}
 		close(c)
 	}()
 
 	select {
-	case <-c:
-		t.Fatalf("Unexpected end of rpc call")
-	case <-time.After(1 * time.Millisecond):
+	case err := <-c:
+		t.Fatalf("Unexpected end of rpc call: %v", err)
+	case <-time.After(10 * time.Millisecond):
 	}
 
 	// Grab the client for our invalid address and close it. This will cause the

